pkg/agent/agent.go

```diff
@@ -319,7 +319,12 @@ func getAgentResource() (*apiV1.ResourceInstance, error) {
 
 	response, err := agent.apicClient.ExecuteAPI(coreapi.GET, agentResourceURL, nil, nil)
 	if err != nil {
-		return nil, err
+		// if agentResourceType is already a standard generic agent resource
+		if agentResourceType == agentTypesMap[agent.cfg.AgentType] {
```

Comment:
or move this up under line 317?

Reply:
this prevents an infinite loop, if an error is hit trying to get a generic agent we return the error an ddon't continuously try